domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Agency.java

```diff
@@ -232,8 +233,11 @@ public AgencyBuilder agencyEmail(@Nullable final String agencyEmail) {
         public EntityBuildResult<?> build() {
             noticeCollection.clear();
 
-            if (agencyName == null || agencyUrl == null || agencyTimezone == null) {
-
+            if (agencyName == null || agencyUrl == null || agencyTimezone == null ||
+                    (agencyId!=null && agencyId.isBlank())) {
+                if (agencyId!=null && agencyId.isBlank()) {
+                    noticeCollection.add(new InvalidAgencyIdNotice(agencyId));
```

Comment:
I'm not sure to understand the logic behind this. An agency ID null or blank is valid if there is only 1 agency in agency.txt. Maybe this should be checked in the use case instead.

Reply:
When building an `Agency` entity, field `agency_id` can be null. As `agency_id` is conditionally required, we check this condition in a use case.

> This field is required when the dataset contains data for multiple transit agencies, otherwise it is optional

However,  if provided, it can not be left blank. Therefore it is considered as non-valid in this implementation.

Reply:
If provided, `agency_id` should not be blank or empty "`   `" or `""`. That's the reason why this check is done in the builder. We only check the presence of a valid `agency_id` (if there is more than one record in `agency.txt` in the related use case `ValidateAgencyIdRequirement`.

Reply:
Thinking more about this, the specification says that `agency_id` can be any sequence of UTF-8 characters. `SPACE` qualifies as a UTF-8 character, so my remark might not be relevant here (http://www.fileformat.info/info/charset/UTF-8/list.htm). 

Leaving this discussion open for further discussion.

Reply:
@lionel-nj I agree with your comment - the intent of the spec is that the agency_id shouldn't be empty or whitespace - it needs to be something visible to the end user.